### DISCUSSION TITLE
Support of wallet-standard encrypt/decrypt

### DISCUSSION THREAD

**Discussion by tanglepay-dev** [phantom/docs#252] [General]
There is two experimental interfaces in wallet-standard for encryption/decryption of arbitrary data using the ed25519 key pairs.

https://github.com/wallet-standard/wallet-standard/blob/master/packages/experimental/features/src/encrypt.ts
https://github.com/wallet-standard/wallet-standard/blob/master/packages/experimental/features/src/decrypt.ts

Can phantom support these two via wallet-adapters?

// New to the Solona ecosystem, sorry if there is an oversight of current implementation.

**Comment by adamdelphantom** [comment]
@tanglepay-dev Sorry for the delay!

Would you mind providing more info about your use case for these interfaces?

**Reply by FrostF0X** [reply]
What about p2p data exchange so there will not be any third-party involved. 
For example passing address from buyer to seller in decentralized marketplace.

**Reply by tanglepay-dev** [reply]
These interfaces will help apps light up messaging scenarios with e2e encryption.

**Comment by 355am** [comment]
This would be amazing! I'm surprised this hasn't been implemented along with `signMessage()`.

Encryption can already be done using the account address (public key) with packages like `tweetnacl`, but Phantom could enable it's users to do the decryption without exposing the private key. Would be very useful and  I'm sure this would extend Phantom's use cases beyond being a "simple wallet".

@adamdelphantom please consider!
cc @FrostF0X @tanglepay-dev

**Comment by NicoBeCodin** [comment]
Hello, I'm is there anything new concerning these features ?

**Reply by adamdelphantom** [reply]
@NicoBeCodin No plans yet. Our product team will continue to consider this for the future.
